integration/src/main/java/scripts/testscripts/EnumerateJobs.java

```diff
@@ -0,0 +1,198 @@
+
+public class EnumerateJobs extends DataRepoTestScriptBase {
+  private static final Logger logger = LoggerFactory.getLogger(EnumerateJobs.class);
+
+  // The test is written so that these can be modified here. The invariant is that the
+  // resulting set of resources can be read in 3 pages where the third page is not full.
+  // Number of resources to create for enumeration
+  private static final int RESOURCE_COUNT = 12;
+  // Page size to use for enumeration paging
+  private static final int PAGE_SIZE = 4;
+  // Roles to grant user on private resource
+  private static final ImmutableList<ControlledResourceIamRole> PRIVATE_ROLES =
+      ImmutableList.of(ControlledResourceIamRole.WRITER, ControlledResourceIamRole.EDITOR);
+
+  private ControlledGcpResourceApi ownerControlledGcpResourceApi;
+  private ReferencedGcpResourceApi ownerReferencedGcpResourceApi;
+  private ResourceApi ownerResourceApi;
+  private ResourceApi readerResourceApi;
+  private Alpha1Api alpha1Api;
+  private List<ResourceMetadata> resourceList;
+  private TestUserSpecification workspaceReader;
+
+  @Override
+  public void doSetup(List<TestUserSpecification> testUsers, WorkspaceApi workspaceApi)
+      throws Exception {
+
+    // initialize workspace
+    super.doSetup(testUsers, workspaceApi);
+
+    TestUserSpecification workspaceOwner = testUsers.get(0);
+
+    // TODO: if we like the alpha1 API for job enumeration, then we can maybe piggyback on
+    //  the EnumerateResources test instead of creating our own set.
+
+    ApiClient ownerApiClient = ClientTestUtils.getClientForTestUser(workspaceOwner, server);
+    ownerControlledGcpResourceApi = new ControlledGcpResourceApi(ownerApiClient);
+    ownerReferencedGcpResourceApi = new ReferencedGcpResourceApi(ownerApiClient);
+    ownerResourceApi = new ResourceApi(ownerApiClient);
+    alpha1Api = new Alpha1Api(ownerApiClient);
+
+    // Create a cloud context for the workspace
+    CloudContextMaker.createGcpCloudContext(getWorkspaceId(), workspaceApi);
+
+    // create the resources for the test
+    logger.info("Creating {} resources", RESOURCE_COUNT);
+    resourceList =
+        ResourceMaker.makeResources(
+            ownerReferencedGcpResourceApi,
+            ownerControlledGcpResourceApi,
+            getWorkspaceId(),
+            getDataRepoSnapshotId(),
+            getDataRepoInstanceName(),
+            RESOURCE_COUNT);
+
+    logger.info("Created {} resources", resourceList.size());
+    logger.info("Cleaning up {} resources", resourceList.size());
+
+    ResourceMaker.cleanupResources(resourceList, ownerControlledGcpResourceApi, getWorkspaceId());
+    logger.info("Cleaned up {} resources", resourceList.size());
+  }
+
+  @Override
+  public void doUserJourney(TestUserSpecification testUser, WorkspaceApi workspaceApi)
+      throws Exception {
+
+    // Case 1: fetch all
+    EnumerateJobsResult fetchall =
+        alpha1Api.enumerateJobs(getWorkspaceId(), null, null, null, null, null, null);
```

Comment:
Our practice has been not to do that since IntelliJ auto-fills them in when you are looking at the code, but I'm happy to check in with folks about that practice.

Reply:
go/java-practices/methods#call-site-literals the java style guide recommends inline comments.